waspc/cli/src/Wasp/Cli/Command/BuildStart.hs

```diff
@@ -0,0 +1,175 @@
+    J.WebApp
+    & toJobExcept (("Building the client failed with exit code: " <>) . show)
+  where
+    webAppDir = buildDir </> Common.webAppRootDirInProjectRootDir
+
+startClient :: SP.Path' SP.Abs (SP.Dir ProjectRootDir) -> JobExcept
+startClient buildDir =
+  runNodeCommandAsJob
+    webAppDir
+    "npm"
+    [ "exec",
+      "vite",
+      "--",
+      "preview", -- `preview` launches vite just as a webserver to the built files
+      "--port",
+      show defaultClientPort,
+      "--strictPort" -- This will make it fail if the port is already in use
+    ]
+    J.WebApp
+    & toJobExcept (("Serving the client failed with exit code: " <>) . show)
+  where
+    webAppDir = buildDir </> Common.webAppRootDirInProjectRootDir
+
+buildServer :: SP.Path' SP.Abs (SP.Dir ProjectRootDir) -> String -> JobExcept
+buildServer buildDir dockerImageName =
+  runProcessAsJob
+    (proc "docker" ["build", "--tag", dockerImageName, SP.fromAbsDir buildDir])
+    J.Server
+    & toJobExcept (("Building the server failed with exit code: " <>) . show)
+
+startServer :: SP.Path' SP.Abs (SP.Dir WaspProjectDir) -> String -> String -> String -> JobExcept
+startServer projectDir clientUrl dockerImageName containerName =
+  ( \chan -> do
+      jwtSecret <- randomAsciiAlphaNum 32 <$> newStdGen
+
+      runProcessAsJob
+        ( proc "docker" $
+            ["run", "--name", containerName, "--rm", "--env-file", envFilePath, "--network", "host"]
+              ++
+              -- We specifically pass this environment variable from the current
+              -- execution to the server container because Prisma will need it,
+              -- and it is not set in the .env file (wasp start complains if so).
+              ["--env", "DATABASE_URL"]
+              ++ toDockerEnvFlags
+                [ ("WASP_WEB_CLIENT_URL", clientUrl),
+                  ("WASP_SERVER_URL", defaultDevServerUrl),
+                  ("JWT_SECRET", jwtSecret)
+                ]
+              ++ [dockerImageName]
+        )
+        J.Server
+        chan
+  )
+    & toJobExcept (("Running the server failed with exit code: " <>) . show)
+  where
+    envFilePath = SP.fromAbsFile $ projectDir </> dotEnvServer
+
+    randomAsciiAlphaNum :: RandomGen g => Int -> g -> String
+    randomAsciiAlphaNum len gen = take len $ filter isAlphaNum $ randoms gen
+      where
+        isAlphaNum c = isAsciiUpper c || isAsciiLower c || isDigit c
+
+    toDockerEnvFlags :: [(String, String)] -> [String]
+    toDockerEnvFlags = concatMap (\(name, value) -> ["--env", name ++ "=" ++ value])
+
+-- | Docker image name unique for the Wasp project with specified path and name.
+makeAppImageName :: SP.Path' SP.Abs (SP.Dir WaspProjectDir) -> String -> String
+makeAppImageName waspProjectDir appName =
+  map toLower $
+    makeAppUniqueId waspProjectDir appName <> "-server"
+
+makeAppContainerName :: SP.Path' SP.Abs (SP.Dir WaspProjectDir) -> String -> String
+makeAppContainerName waspProjectDir appName =
```

Comment:
Would it make sense to define this one via `makeAppImageName`? Do we want to maintain this property that this one is just the image name + `-container`, or is that an "accident"?